### PR TITLE
Create Pathlike objects

### PR DESCRIPTION
**Pull request by fyellin**
First step towards generic handling of Path-like objects.  As a test, I've made ldpool and furnsh use these.  So they can accept as their file argument: strings, bytes, or Paths.

As part of this, also moved SwigCallbacks into its own file and renamed it.

Also added tests for SpiceCell, and simplified the word that Swig needs to use to access them.

Eventually, I'll need a list of all methods that need to be rewritten.  Or I hope that it'll be simple enough what I've done

### CODE REVIEW

**Comment by rfrenchseti** [comment]
Can you add new tests for furnsh and ldpool that use pathlib?

**Comment by fyellin** [comment]
Added tests to ldpool and furnsh.  Made it easy to use an @annotation so any file test can be modified to test strings, byte strings, and filepaths.

I'm slightly unhappy with my calling my variable name "modifier".  Any better name?  The test is run three times, and each time its bound to a different function.

File: cspyce/__init__.py [hunk 1]
```diff
@@ -137,7 +137,7 @@
         print("Set environment variable 'CSPICE_DEVELOPMENT' to '1' to ignore this error")
         raise err
 
-from cspyce.spice_cell import SPICE_CELL_INT, SPICE_CELL_DOUBLE, SpiceCell
+from cspyce.spice_cell import *
```
Comment (rfrenchseti) [inline cspyce/__init__.py:140]: Can we get away from `import *`? It screws up IDEs trying to track symbols.

File: unittests/test_f_j.py [hunk 2]
```diff
@@ -110,8 +111,10 @@ def test_frmnam():
     assert cs.frmnam(13000) == "ITRF93"
 
 
-def test_furnsh():
-    cs.furnsh(CoreKernels.testMetaKernel)
+@checking_pathlike_filename_variants("modifier")
```
Comment (rfrenchseti) [inline unittests/test_f_j.py:114]: How about something like `path_type_variant`?

**Review by rfrenchseti — CHANGES_REQUESTED** [review]

File: unittests/test_f_j.py (hunk 2 above)
Comment (fyellin) [reply unittests/test_f_j.py]: Done.  I like that.

**Comment by fyellin** [comment]
Hold off approving.  I just found an C function that might simplify this code.  Want to check.

**Comment by fyellin** [comment]
Change made.  Everything works.  No need to callback into Python.

I was not expecting to find the function I had been looking for in PyUnicode_.

**Review by rfrenchseti — APPROVED** [review]

**Pull request merged**